components/application-connectivity-validator/internal/validationproxy/handler_test.go

```diff
@@ -0,0 +1,232 @@
+package validationproxy
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	applicationName = "test-application"
+
+	group  = "group"
+	tenant = "tenant"
+
+	eventServicePathPrefix = "/test-application/v1/events"
+	appRegistryPathPrefix  = "/test-application/v1/metadata"
+)
+
+type event struct {
+	Title string `json:"title"`
+}
+
+func TestProxyHandler_ProxyAppConnectorRequests(t *testing.T) {
+
+	eventServiceHandler := mux.NewRouter()
+	eventServiceServer := httptest.NewServer(eventServiceHandler)
+	eventServiceHost := strings.TrimPrefix(eventServiceServer.URL, "http://")
+
+	appRegistryHandler := mux.NewRouter()
+	appRegistryServer := httptest.NewServer(appRegistryHandler)
+	appRegistryHost := strings.TrimPrefix(appRegistryServer.URL, "http://")
+
+	testCases := []struct {
+		caseDescription string
+		tenant          string
+		group           string
+		certInfoHeader  string
+		expectedStatus  int
+	}{
+		{
+			caseDescription: "Application without group and tenant",
+			certInfoHeader: `Hash=f4cf22fb633d4df500e371daf703d4b4d14a0ea9d69cd631f95f9e6ba840f8ad;Subject="CN=test-application,OU=OrgUnit,O=Organization,L=Waldorf,ST=Waldorf,C=DE";` +
+				`URI=,By=spiffe://cluster.local/ns/kyma-integration/sa/default;` +
+				`Hash=6d1f9f3a6ac94ff925841aeb9c15bb3323014e3da2c224ea7697698acf413226;Subject="";` +
+				`URI=spiffe://cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account`,
+			expectedStatus: http.StatusOK,
+		},
+		{
+			caseDescription: "Application without group and tenant and Common Name is invalid",
```

Comment:
I would also consider changing a description of this use case to something like : "Application in the local context with invalid Common Name"